app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -583,18 +583,32 @@ class NimbusStatusValidationMixin:
     def validate(self, data):
         data = super().validate(data)
 
-        restrictive_statuses = {
+        restrictions = {
             "status": NimbusConstants.STATUS_ALLOWS_UPDATE,
             "publish_status": NimbusConstants.PUBLISH_STATUS_ALLOWS_UPDATE,
         }
+        update_exempt_fields = NimbusExperiment.STATUS_UPDATE_EXEMPT_FIELDS
 
         if self.instance:
-            for status_field, restricted_statuses in restrictive_statuses.items():
-                current_status = getattr(self.instance, status_field)
-                is_locked = current_status not in restricted_statuses
-                modifying_fields = set(data.keys()) - set(
-                    NimbusExperiment.STATUS_UPDATE_EXEMPT_FIELDS
-                )
+            restrictive_statuses = set()
+            exempt_fields = set()
+            if self.instance.is_rollout:
+                fields = ["all", "rollouts"]
```

Comment:
I would have initialized fields up, before if as `fields = ["all"]` and later ` fields.append("experiment") if self.instance.is_rollout else fields.append("rollout")`

```python
fields = ["all"]
fields.append("experiment") if self.instance.is_rollout else fields.append("rollout")

Just a suggestion, to avoid duplicate field initialization 
```